core/src/main/java/hivemall/utils/math/StatsUtils.java

```diff
@@ -189,4 +193,83 @@ public static double hellingerDistance(@Nonnull final RealVector mu1,
         return 1.d - numerator / denominator;
     }
 
+    /**
+     * @param observed mean vector whose value is observed
+     * @param expected mean vector whose value is expected
+     * @return chi2 value
+     */
+    public static double chiSquare(@Nonnull final double[] observed,
+            @Nonnull final double[] expected) {
+        Preconditions.checkArgument(observed.length == expected.length);
+
+        double sumObserved = 0.d;
+        double sumExpected = 0.d;
+
+        for (int ratio = 0; ratio < observed.length; ++ratio) {
+            sumObserved += observed[ratio];
+            sumExpected += expected[ratio];
+        }
+
+        double var15 = 1.d;
+        boolean rescale = false;
+        if (Math.abs(sumObserved - sumExpected) > 1.e-5) {
+            var15 = sumObserved / sumExpected;
+            rescale = true;
+        }
+
+        double sumSq = 0.d;
+
+        for (int i = 0; i < observed.length; ++i) {
+            double dev;
+            if (rescale) {
+                dev = observed[i] - var15 * expected[i];
+                sumSq += dev * dev / (var15 * expected[i]);
+            } else {
+                dev = observed[i] - expected[i];
+                sumSq += dev * dev / expected[i];
+            }
+        }
+
+        return sumSq;
+    }
+
+    /**
+     * @param observed means vector whose value is observed
+     * @param expected means vector whose value is expected
+     * @return p value
+     */
+    public static double chiSquareTest(@Nonnull final double[] observed,
+            @Nonnull final double[] expected) {
+        ChiSquaredDistribution distribution = new ChiSquaredDistribution(null,
```

Comment:
yes, `chiSequares` is an optional impl for reusing
change current impl to make it use `Well19937c` as random generator, following original impl by common-math